### DISCUSSION TITLE
Visaul Studio 2015 Intellisense fails when including ryaml_all.hpp

### DISCUSSION THREAD

**Discussion by jslezosky** [biojppm/rapidyaml#236] [Q&A] [answered]
Any idea why Intellisense in Visual Studio 2015 fails when I include ryaml_all.hpp in a separate Visaul Studio solution?
For example, Intellisense doesn't recognize locally defined types, or ryml types.

Solution builds and executes fine, but Intellisense is confused.

**Comment by biojppm** [comment, marked as answer]
Maybe some of the SFINAE macros in c4core confuse the Visual Studio parser. I can't help much here, as I don't use this much.

If it helps, the Intellisense parser in VS has improved by leaps and bounds since VS2015, so one possible workaround is  installing VS2019 or maybe even VS2022 and use it with the VS2015 solution/toolset (but don't let the more recent VS update the solution, which it will ask for).

Or -- if you must really use the old VS2015 -- and are inclined to figure this out, you can try use the preprocessor to do a binary search for the code that prompts the crash : by using `#if 0` to disable one half of the `ryml_all.hpp` file and click to rescan the solution, you will know which half is causing Intellisense to choke. Then refine from there, by disabling one half of the problematic half, and so on.

**Reply by jslezosky** [reply]
You are the man! IntelliSense does work in 2019!
